Sources/PointFree/Account/Invoices.swift

```diff
@@ -0,0 +1,302 @@
+          title: "Payment history"
+        )
+    }
+)
+
+let invoiceResponse =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
+    <<< requireStripeSubscription
+    <<< filterMap(
+      over3(fetchInvoice) >>> sequence3 >>> map(require3),
+      or: redirect(
+        to: .account(.invoices(.index)),
+        headersMiddleware: flash(
+          .error,
+          """
+          We had some trouble loading your invoice! Please try again later.
+          If the problem persists, please notify <support@pointfree.co>.
+          """
+        )
+      )
+    )
+    <| writeStatus(.ok)
+    >-> map(lower)
+    >>> respond(
+      view: invoiceView,
+      layoutData: { subscription, currentUser, invoice in
+        SimplePageLayoutData(
+          currentUser: currentUser,
+          data: (subscription, currentUser, invoice),
+          style: .minimal,
+          title: "Invoice"
+        )
+    }
+)
+
+private func fetchInvoice(id: Stripe.Invoice.Id) -> IO<Stripe.Invoice?> {
+  return AppEnvironment.current.stripe.fetchInvoice(id)
+    .run
+    .map(^\.right)
+}
+
+private func fetchInvoices<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Stripe.ListEnvelope<Stripe.Invoice>, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Stripe.Subscription, A>, Data> {
+
+    return { conn in
+      let subscription = conn.data.first
+
+      return AppEnvironment.current.stripe.fetchInvoices(subscription.customer)
+        .withExcept(notifyError(subject: "Couldn't load invoices"))
+        .run
+        .flatMap {
+          switch $0 {
+          case let .right(invoices):
+            return conn.map(const(subscription .*. invoices .*. conn.data.second))
+              |> middleware
+          case .left:
+            return conn |> redirect(
+              to: .account(.index),
+              headersMiddleware: flash(
+                .error,
+                """
+                We had some trouble loading your invoices! Please try again later.
+                If the problem persists, please notify <support@pointfree.co>.
+                """
+              )
+            )
+          }
+      }
+    }
+}
+
+// MARK: Views
+
+let invoicesView = View<(Stripe.Subscription, Stripe.ListEnvelope<Stripe.Invoice>, Database.User)> { subscription, invoicesEnvelope, currentUser -> Node in
+
+  gridRow([
+    gridColumn(sizes: [.mobile: 12, .desktop: 8], [style(margin(leftRight: .auto))], [
+      div([`class`([Class.padding([.mobile: [.all: 3], .desktop: [.all: 4]])])],
+          titleRowView.view(unit)
+            <> invoicesRowView.view(invoicesEnvelope)
+      )
+      ])
+    ])
+}
+
+private let titleRowView = View<Prelude.Unit> { _ in
+  gridRow([`class`([Class.padding([.mobile: [.bottom: 2]])])], [
+    gridColumn(sizes: [.mobile: 12], [
+      div([
+        h1([`class`([Class.pf.type.title2])], ["Payment history"])
```

Comment:
you should use `responseTitle2` so that it gets smaller on mobile.

I should deprecate all the `titleX` classes.

Reply:
Ah coo. I copy-pasted this from another screen.